axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/Checksum.java

```diff
@@ -32,10 +47,25 @@ public Checksum update(byte[] bytes) {
         return this;
     }
 
+    /**
+     * Update the checksum with the specified data.
+     *
+     * @param buffer   the buffer that provides the data to be used to calculate the checksum
+     * @param position the initial position in the buffer of the data to be used to calculate the checksum
+     * @param size     the size of the data to be used to calculate checksum
+     * @return this instance, for fluent api
+     */
     public Checksum update(ByteBuffer buffer, int position, int size) {
-        byte[] bytes = new byte[size];
-        ((ByteBuffer) buffer.duplicate().position(position)).get(bytes);
-            crc32.update( bytes);
+        ByteBuffer b = (ByteBuffer) buffer.duplicate().position(position);
+
+        int rem = size;
+        while (rem > 0) {
+            int batchSize = Math.min(batch, rem);
+            byte[] bytes = new byte[batchSize];
```

Comment:
Just keep a single array of size `batch`, which is _constant_, instead of generating high amounts of garbage on the heap!